webhook/webhook.go

```diff
@@ -235,15 +236,14 @@ func SetDefaults(ctx context.Context) ResourceDefaulter {
 
 func configureCerts(ctx context.Context, client kubernetes.Interface, options *ControllerOptions) (*tls.Config, []byte, error) {
 	apiServerCACert, err := getAPIServerExtensionCACert(client)
-	if err != nil {
+	if (options.ClientAuth >= tls.VerifyClientCertIfGiven) && err != nil {
```

Comment:
Two questions:

Why is this a `>=` check? Are we checking for either `VerifyClientCertIfGiven` or `RequireAndVerifyClientCert`? If so I suggest changing this to an OR.

Why do we only check the error value if the ClientAuth option is set?

Reply:
I don't know enough about Kubernetes webhook certs to say for sure. Paging @mattmoor :smile: 

/cc @mattmoor

Reply:
This is @vaikas-google territory

Reply:
right, that's the point that i try to do, and push some new changes, PTAL